lib/modules/manager/pixi/schema.ts

```diff
@@ -1,10 +1,196 @@
+)
+  .catch({})
+  .transform(collectNamedPackages);
+
+const CondaDependency = z
+  .union([
+    z.string().transform((version) => ({ version, channel: undefined })),
+    z.object({ version: z.string(), channel: z.optional(z.string()) }),
+  ])
+  .transform(({ version, channel }) => {
+    return {
+      currentValue: version,
+      versioning: condaVersion.id,
+      datasource: CondaDatasource.id,
+      depType: 'dependencies',
+      channel,
+    } satisfies PixiPackageDependency;
+  });
+
+const CondaDependencies = LooseRecord(z.string(), CondaDependency)
+  .catch({})
+  .transform(collectNamedPackages);
+
+const Targets = LooseRecord(
+  z.string(),
+  z.object({
+    dependencies: z.optional(CondaDependencies).default({}),
+    'pypi-dependencies': z.optional(PypiDependencies).default({}),
+  }),
+).transform((val) => {
+  const conda: PixiPackageDependency[] = [];
+  const pypi: PixiPackageDependency[] = [];
+  for (const value of Object.values(val)) {
+    pypi.push(...value['pypi-dependencies']);
+
+    conda.push(...value.dependencies);
+  }
+
+  return { pypi, conda };
+});
+
+const Project = z.object({
+  channels: z.array(Channel).default([]),
+});
+
+const DependenciesMixin = z
+  .object({
+    dependencies: z.optional(CondaDependencies).default({}),
+    'pypi-dependencies': z.optional(PypiDependencies).default({}),
+    target: z.optional(Targets).default({}),
+  })
+  .transform(
+    (
+      val,
+    ): { pypi: PixiPackageDependency[]; conda: PixiPackageDependency[] } => {
+      return {
+        conda: [...val.dependencies, ...val.target.conda],
+        pypi: [...val['pypi-dependencies'], ...val.target.pypi],
+      };
+    },
+  );
+
+const Features = LooseRecord(
+  z.string(),
+  z
+    .object({
+      channels: z.array(Channel).optional(),
+    })
+    .and(DependenciesMixin),
+)
+  .default({})
+  .transform(
+    (
+      features,
+    ): {
+      conda: PixiPackageDependency[];
+      pypi: PixiPackageDependency[];
+    } => {
+      const pypi: PixiPackageDependency[] = [];
+      const conda: PixiPackageDependency[] = [];
+
+      for (const feature of Object.values(features)) {
+        conda.push(
+          ...feature.conda.map((item) => {
+            return {
+              ...item,
+              channels: feature.channels,
+            };
+          }),
+        );
+
+        pypi.push(...feature.pypi);
+      }
+
+      return { pypi, conda };
+    },
+  );
 
 /**
- * config of `pixi.toml` of `tool.pixi` of `pyproject.toml`
+ * `$` of `pixi.toml` or `$.tool.pixi` of `pyproject.toml`
  */
-export const PixiConfigSchema = z.object({});
+export const PixiConfigSchema = z
+  .object({ feature: Features })
+  .and(DependenciesMixin)
+  .and(
+    z.union([
+      z
+        .object({
+          workspace: Project,
+        })
+        .transform((val) => {
+          return { project: val.workspace };
+        }),
+      z.object({
+        project: Project,
+      }),
```

Comment:
Can really be there only one of `workspace` and `project`. Here AFAIK `workspace` takes precedence.

Reply:
I don't understance this comment

Reply:
You have an union here which checks for the `workspace` and `project`  key. If there is a object like 

```json
{
  "project": { ... },
  "workspace":  { ... }
}
```
as you use a transform function to rewrite `workspace` to `project`, AFAIK in case both are provided the [`project`](https://github.com/renovatebot/renovate/pull/34646/files/9b566064eb49af889ad24e1f343d7e4e409e5277#diff-f86ebf948382b4c42ce9cff6f81083c33fa9fd2a21b0d79624433450d6d114fdR190) will be dropped

Reply:
this is not a valid pixi config, it can be a `project` or a `workspace`

Reply:
pixi will refuse to load it:

pixi.toml

```toml
[workspace]
name = '1'
channels = []
platforms = []

[project]
name = '1'
channels = []
platforms = []
```

but each part is a valid pixi.toml

Reply:
Should I add a manually check that `workspace` and `project` are not both provided?

Reply:
K, then it make sense. I do not understand the logic why Pixi is doing this, but this is not relevant for this implementation

Reply:
application vs library